File: daq/slc/src/base/src/DataSender.cc
#include "DataSender.h"

#include "StringUtil.h"

using namespace Belle2;

const std::string DataSender::getSQLFields() const throw()
{
  return  ", script text, port smallint, event_size int";
}

const std::string DataSender::getSQLLabels() const throw()
{
  return ", script, port, event_size";
}

const std::string DataSender::getSQLValues() const throw()
{
  return  Belle2::form(", '%s', %d, %d", getScript().c_str(),
                       getPort(), getEventSize());
}

File: daq/slc/tools/javalib/b2rc/src/b2rc/core/DataSender.java
package b2rc.core;

public class DataSender extends SCObject {

	public static String TAG = "data_sender";

	private String _script;
	private String _host;
	private short _port;
	private int _event_size;

	public String getScript() {
		return _script;
	}

	public String getHost() {
		return _host;
	}

	public short getPort() {
		return _port;
	}

	public int getEventSize() {
		return _event_size;
	}

	public void setScript(String script) {
		_script = script;
	}

	public void setHost(String host) {
		_host = host;
	}

	public void setPort(short port) {
		_port = port;
	}

	public void setEventSize(int size) {
		_event_size = size;
	}

	@Override
	public String getTag() {
		return TAG;
	}

	@Override
	public String getSQLFields() {
		return ", script text, port smallint, event_size int";
	}

	@Override
	public String getSQLLabels() {
		return ", script, port, event_size";
	}

	@Override
	public String getSQLValues() {
		return ", '" + _script + "', " + _port + ", "
				+ _event_size;
	}

}
